### DISCUSSION TITLE
Vertical mode

### DISCUSSION THREAD

**Discussion by KiSa04** [j-kaltes/Juggluco#172] [Ideas]
It would be nice to introduce a vertical mode option. It's probably not very relevant for most people, but using a flip phone is harder when the app is horizontal :p

![image](https://github.com/j-kaltes/Juggluco/assets/55666909/050ecf0a-b1f6-46e6-bf59-81c606083333)

**Comment by j-kaltes** [comment]
Is the app only shown in one half? The fold phone emulator of Android studio shows Juggluco on both sites:
![fold2](https://github.com/j-kaltes/Juggluco/assets/72013079/ed5f7dc1-f101-40e2-9948-549bf2e1d1c1)
To me it seems that when the two sites are below each other, portrait mode seems to be even more inappropriate to display the glucose curve. In contrast, when both sites are placed besides each other, in landscape mode, you can get an overview of a larger time period.

**Reply by thegrobi** [reply]
Often, I had a situations in which portrait mode is extremely annoying:

Bicycle has smartphone as navi in the holder (upright).
Low blood sugar after 60 km due to exertion.

Ate something and entered 2 BE in Juggluco. Not so easy when Juggluco remains in forced widescreen mode.

I try to use XDrip+ only for doing my entries easier than in juggluco, but to have one more app on phone only for entries is not the best soloution.

**Reply by Puderzo** [reply]
I'm running Juggluco on a OnePlus Open (as a follower to my son), which is a fold-type phone, but whose "opened" format is still marginally portrait (1.0758:1). The Galaxy Z Fold series are somewhat the same (aspect ratio 20.9:18), but with a bit narrower portrait compared to the Open. Both would be wide enough when opened (almost square aspect ratio) to perfectly show the graph also in portrait mode, and feel very awkward to turn sideways just to look at the same "square" in a different angle.

When using Juggluco on Wear OS (as my son does), he's not forced to use it in landscape (this is on a Galaxy Watch 6 LTE), and the graph is still readable even on the very small round screen (albeit needs a bit of scrolling if you want to see more than the latest, but you can always zoom out to see more), so I don't think a larger phone with better resolution could be much worse in terms of amount of information visible.

But really, most of the time that we actually open the Juggluco app, and are not just glancing at the value on a watchface or a widget, we just want to either see why it's not showing values (is it Sensor Error or No connection), connect a new sensor, or toggle the _Direct sensor-watch connection_-setting (as he's using the watch when playing sports for example, but using the watch as collector drains its battery quite fast, so mostly we keep the Libre3 connected to the phone).

We're sending the glucose value onwards from Juggluco to xDrip+, and from there to AAPS (we used to do straight from Juggluco to AAPS, but AAPS got confused by the 1-minute readings so we had to filter it through xDrip+, but I still use both on my phone to be able to judge what the "real" glucose level is). This means that we rarely look at the actual graph in Juggluco longer back than the last hour or so, mainly to see if it's falling or rising very fast or some such; older values than that we look at in AAPS and/or Nightscout, where we then also see the pumped values etc. from the Omnipod Dash he uses, and where it doesn't matter that the value is not as "immediate" as it is in Juggluco. My son never looks at the graph (in neither Juggluco nor AAPS); he just uses the toggle between watch and phone.

For this reason, a toggle in settings to disable the forced landscape would be very much appreciated, for those of us who don't use the graph as much as the rest of the functionality. And when we do want to look at the graph (if using a phone other than the Open), we could just turn it sideways and the auto-rotate function would give us the graph in a landscape view. Best of both worlds, a win-win.

**Comment by zyt** [comment]
I had the similar question/observation and I don't want to open another discussion thread. Most of the phone holders (bike, car, etc.) fix the phone in vertical model. It makes it very hard to enter any Amounts or to read the glucose values from the app when the phone is physically fixed in portrait mode inside a holder. Would it be possible to at least pinpoint where in the code the landscape mode is forced? Or perhaps allow portrait mode in the settings of the app ? I can see that `AndroidManifest.xml` of the apk is explicitly set with `android:screenOrientation="reverseLandscape"`.

**Comment by j-kaltes** [comment]
Android is made in such a way then when portrait mode is possible, you end up staying constantly in portrait mode. Displaying the glucose graph in portrait mode, as is done with nearly all glucose apps, is ridiculous. I don’t believe that it is really so difficult to enter a number with the phone in landscape mode. I never had any trouble with it.

**Reply by KiSa04** [reply]
There are phones where scrolling is impossible, making it hard to navigate. I'll attach my example bellow

I do know that this is probably a very niche feature, but it's worth suggesting

https://github.com/user-attachments/assets/c98ab57c-0dc2-41bb-8506-f181e4800c28

**Reply by erizzo** [reply]
> Android is made in such a way then when portrait mode is possible, you end up staying constantly in portrait mode.

Can you explain what you mean? There are plenty of apps that can switch between portrait and landscape as you rotate the device. Google Maps, Amazon, Camera, etc.

Consider this another vote in favor of a way to use portrait - it's the biggest reason I don't use Juggluco's UI for anything after scanning to start a sensor.

**Reply by j-kaltes** [reply]
In landscape mode you can still see something. In portrait you would see absolutely nothing on that screen. You could as well only ook at the glucose notification in portrait mode.

In my bike frontbag my tablet fits only in landscape mode. I have tried to navigate that way using Komoot. You have to turn off autorotate otherwise it constantly changes orientation. With every notification or even unlocking the phone, it switches back to Portrait mode and stays that way.

**Reply by GuillaumeGomez** [reply]
Late answer: why not providing an option to enable to lock the screen orientation in any given direction (portrait or landscape)?

Like that your usage still works and other users too.

**Reply by drudeboy** [reply]
No portrait orientation on modern smartphones is just silly. Why is there this stubborness about it?

**Reply by GuillaumeGomez** [reply]
Because everyone has different tastes and allowing people to chose what they want for themselves seems like an acceptable option?

**Reply by drudeboy** [reply]
I agree! I want portrait mode as an option, not locked to landscape

**Comment by drudeboy** [comment]
Sort of portrait mode with Bubble feature on latest pixel update 
<img width="1344" height="2992" alt="Screenshot_20260816-152241" src="https://github.com/user-attachments/assets/045ade36-0bcd-4b8a-bb3f-9a48e9c9de9e" />
